src/main/java/org/datavyu/plugins/nativeosx/NativeOSXMediaPlayer.java

```diff
@@ -153,56 +158,56 @@ protected synchronized void playerSetVolume(float volume) throws MediaException
 
   @Override
   protected double playerGetDuration() throws MediaException {
-    if (duration <= -1) {
-      return duration = nativePlayerCanvas.getDuration(id) / 1000F;
-    }
-    return duration;
+    return (mediaPlayer.getDuration(id) / 1000.0);
   }
 
   @Override
   protected void playerSeek(double streamTime) throws MediaException {
     if (!seeking) {
       seeking = true;
       EventQueue.invokeLater(() -> {
-        logger.info("Seeking to position: " + streamTime + " Is playing: " + isPlaying());
         boolean wasPlaying = isPlaying();
         prevRate = playerGetRate();
         if (isPlaying()) {
-          nativePlayerCanvas.stop(id);
+          mediaPlayer.stop(id);
         }
         if (!wasPlaying || prevRate >= 0 && prevRate <= 8) {
-          playerSeek(streamTime, PRECISE_SEEK_FLAG);
+          logger.info("Precise seek to position: " + streamTime);
+          playerSeek( streamTime * 1000, PRECISE_SEEK_FLAG);
         } else if(prevRate < 0 && prevRate > - 8) {
-          playerSeek(streamTime, MODERATE_SEEK_FLAG);
+          logger.info("Moderate seek to position: " + streamTime);
+          playerSeek(streamTime * 1000, MODERATE_SEEK_FLAG);
         } else {
-          playerSeek(streamTime, NORMAL_SEEK_FLAG);
+          logger.info("Seek to position: " + streamTime);
```

Comment:
Switch log level info -> debug

Rational:  Seeks could happen frequently and I don't want to clutter the logs.

Reply:
Done!